fairscale/nn/wrap/auto_wrap.py

```diff
@@ -34,26 +88,25 @@ def enable_wrap(module_blocklist: Optional[List] = None, **wrapper_kwargs: Any)
         with enable_wrap(**params):
             # Wraps layer in FSDP by default if within context
             self.l1 = wrap(torch.nn.Linear(5, 5))
-            # Wraps children modules by default based on min_num_params
-            self.l2 = auto_wrap(TransformerBlock(), min_num_params=1e8)
+            # Wraps children modules based on a different min_num_params
+            my_should_wrap = functools.partial(should_wrap, min_num_params=1e7)
+            self.l2 = auto_wrap(TransformerBlock(), shuold_wrap=my_should_wrap)
 
     Args:
-        module_blocklist: List of additional Module Classes to not wrap when
-            using :func:`auto_wrap`. This is useful to exclude unsupported
-            modules when wrapping recursively.
-        **wrapper_kwargs: Configuration settings that will be passed to all ``wrap``
+        should_wrap (Callable, Optional):
+            Custom function to control how to do :func:`auto_wrap`. This is
+            useful to exclude unsupported modules or wrap based on sizes when
+            wrapping recursively.
+            (default: :func:`default_should_wrap_policy`)
```

Comment:
it's not totally clear from this description if this applies to `wrap` layers or just `auto_wrap`. For example:
```python
with enable_wrap(should_wrap=lambda _, module, _: False):
    l1 = wrap(nn.Linear(1, 1))
```
In this case the Linear would be wrapped, correct?

Assuming yes, then perhaps update this description to something like:
```
Custom function to control how to do :func:`auto_wrap`. This is
useful to exclude unsupported modules or wrap based on sizes when
wrapping recursively. Note: modules annotated with :func:`wrap`
ignore this policy and will always be wrapped
(default: :func:`default_should_wrap_policy`)
```

Given the above, I'm also wondering if it'd be clear to rename `should_wrap` to `auto_wrap_policy`. What do you think?

Reply:
good idea. done.